// @atproto/api/dist/client/types/com/atproto/repo/applyWrites.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateDeleteResult = exports.isDeleteResult = exports.validateUpdateResult = exports.isUpdateResult = exports.validateCreateResult = exports.isCreateResult = exports.validateDelete = exports.isDelete = exports.validateUpdate = exports.isUpdate = exports.validateCreate = exports.isCreate = exports.toKnownErr = exports.InvalidSwapError = void 0;
/**
 * GENERATED CODE - DO NOT MODIFY
 */
const xrpc_1 = require("@atproto/xrpc");
const util_1 = require("../../../../util");
const lexicons_1 = require("../../../../lexicons");
class InvalidSwapError extends xrpc_1.XRPCError {
    constructor(src) {
        super(src.status, src.error, src.message, src.headers, { cause: src });
    }
}
exports.InvalidSwapError = InvalidSwapError;
function toKnownErr(e) {
    if (e instanceof xrpc_1.XRPCError) {
        if (e.error === 'InvalidSwap')
            return new InvalidSwapError(e);
    }
    return e;
}
exports.toKnownErr = toKnownErr;
function isCreate(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'com.atproto.repo.applyWrites#create');
}
exports.isCreate = isCreate;
function validateCreate(v) {
    return lexicons_1.lexicons.validate('com.atproto.repo.applyWrites#create', v);
}
exports.validateCreate = validateCreate;
function isUpdate(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'com.atproto.repo.applyWrites#update');
}
exports.isUpdate = isUpdate;
function validateUpdate(v) {
    return lexicons_1.lexicons.validate('com.atproto.repo.applyWrites#update', v);
}
exports.validateUpdate = validateUpdate;
function isDelete(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'com.atproto.repo.applyWrites#delete');
}
exports.isDelete = isDelete;
function validateDelete(v) {
    return lexicons_1.lexicons.validate('com.atproto.repo.applyWrites#delete', v);
}
exports.validateDelete = validateDelete;
function isCreateResult(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'com.atproto.repo.applyWrites#createResult');
}
exports.isCreateResult = isCreateResult;
function validateCreateResult(v) {
    return lexicons_1.lexicons.validate('com.atproto.repo.applyWrites#createResult', v);
}
exports.validateCreateResult = validateCreateResult;
function isUpdateResult(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'com.atproto.repo.applyWrites#updateResult');
}
exports.isUpdateResult = isUpdateResult;
function validateUpdateResult(v) {
    return lexicons_1.lexicons.validate('com.atproto.repo.applyWrites#updateResult', v);
}
exports.validateUpdateResult = validateUpdateResult;
function isDeleteResult(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'com.atproto.repo.applyWrites#deleteResult');
}
exports.isDeleteResult = isDeleteResult;
function validateDeleteResult(v) {
    return lexicons_1.lexicons.validate('com.atproto.repo.applyWrites#deleteResult', v);
}
exports.validateDeleteResult = validateDeleteResult;
