// @atproto/api/dist/client/types/app/bsky/graph/defs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateRelationship = exports.isRelationship = exports.validateNotFoundActor = exports.isNotFoundActor = exports.validateListViewerState = exports.isListViewerState = exports.REFERENCELIST = exports.CURATELIST = exports.MODLIST = exports.validateStarterPackViewBasic = exports.isStarterPackViewBasic = exports.validateStarterPackView = exports.isStarterPackView = exports.validateListItemView = exports.isListItemView = exports.validateListView = exports.isListView = exports.validateListViewBasic = exports.isListViewBasic = void 0;
const util_1 = require("../../../../util");
const lexicons_1 = require("../../../../lexicons");
function isListViewBasic(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'app.bsky.graph.defs#listViewBasic');
}
exports.isListViewBasic = isListViewBasic;
function validateListViewBasic(v) {
    return lexicons_1.lexicons.validate('app.bsky.graph.defs#listViewBasic', v);
}
exports.validateListViewBasic = validateListViewBasic;
function isListView(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'app.bsky.graph.defs#listView');
}
exports.isListView = isListView;
function validateListView(v) {
    return lexicons_1.lexicons.validate('app.bsky.graph.defs#listView', v);
}
exports.validateListView = validateListView;
function isListItemView(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'app.bsky.graph.defs#listItemView');
}
exports.isListItemView = isListItemView;
function validateListItemView(v) {
    return lexicons_1.lexicons.validate('app.bsky.graph.defs#listItemView', v);
}
exports.validateListItemView = validateListItemView;
function isStarterPackView(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'app.bsky.graph.defs#starterPackView');
}
exports.isStarterPackView = isStarterPackView;
function validateStarterPackView(v) {
    return lexicons_1.lexicons.validate('app.bsky.graph.defs#starterPackView', v);
}
exports.validateStarterPackView = validateStarterPackView;
function isStarterPackViewBasic(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'app.bsky.graph.defs#starterPackViewBasic');
}
exports.isStarterPackViewBasic = isStarterPackViewBasic;
function validateStarterPackViewBasic(v) {
    return lexicons_1.lexicons.validate('app.bsky.graph.defs#starterPackViewBasic', v);
}
exports.validateStarterPackViewBasic = validateStarterPackViewBasic;
/** A list of actors to apply an aggregate moderation action (mute/block) on. */
exports.MODLIST = 'app.bsky.graph.defs#modlist';
/** A list of actors used for curation purposes such as list feeds or interaction gating. */
exports.CURATELIST = 'app.bsky.graph.defs#curatelist';
/** A list of actors used for only for reference purposes such as within a starter pack. */
exports.REFERENCELIST = 'app.bsky.graph.defs#referencelist';
function isListViewerState(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'app.bsky.graph.defs#listViewerState');
}
exports.isListViewerState = isListViewerState;
function validateListViewerState(v) {
    return lexicons_1.lexicons.validate('app.bsky.graph.defs#listViewerState', v);
}
exports.validateListViewerState = validateListViewerState;
function isNotFoundActor(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'app.bsky.graph.defs#notFoundActor');
}
exports.isNotFoundActor = isNotFoundActor;
function validateNotFoundActor(v) {
    return lexicons_1.lexicons.validate('app.bsky.graph.defs#notFoundActor', v);
}
exports.validateNotFoundActor = validateNotFoundActor;
function isRelationship(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'app.bsky.graph.defs#relationship');
}
exports.isRelationship = isRelationship;
function validateRelationship(v) {
    return lexicons_1.lexicons.validate('app.bsky.graph.defs#relationship', v);
}
exports.validateRelationship = validateRelationship;
