// @atproto/api/dist/client/types/app/bsky/embed/video.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateView = exports.isView = exports.validateCaption = exports.isCaption = exports.validateMain = exports.isMain = void 0;
const util_1 = require("../../../../util");
const lexicons_1 = require("../../../../lexicons");
function isMain(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        (v.$type === 'app.bsky.embed.video#main' ||
            v.$type === 'app.bsky.embed.video'));
}
exports.isMain = isMain;
function validateMain(v) {
    return lexicons_1.lexicons.validate('app.bsky.embed.video#main', v);
}
exports.validateMain = validateMain;
function isCaption(v) {
    return ((0, util_1.isObj)(v) &&
        (0, util_1.hasProp)(v, '$type') &&
        v.$type === 'app.bsky.embed.video#caption');
}
exports.isCaption = isCaption;
function validateCaption(v) {
    return lexicons_1.lexicons.validate('app.bsky.embed.video#caption', v);
}
exports.validateCaption = validateCaption;
function isView(v) {
    return ((0, util_1.isObj)(v) && (0, util_1.hasProp)(v, '$type') && v.$type === 'app.bsky.embed.video#view');
}
exports.isView = isView;
function validateView(v) {
    return lexicons_1.lexicons.validate('app.bsky.embed.video#view', v);
}
exports.validateView = validateView;
